hydra/redisgraph_demo1.py

```diff
@@ -0,0 +1,258 @@
+"""
+*Readme for this file*:
+ It contains a graph which contain the api structure of api doc
+ It design with the help of endpoints and its operation
+ Design:
+  url or id => endpoints[] => collection(get) and classes(put or post)
+                             ((if collection then => classes=> members(get))
+                             elif classes then =>members(get))
+
+
+"""
+
+
+import redis
+from redisgraph import Graph, Node, Edge
+import urllib
+import json
+import re
+from re import compile as regex
+from httplib2 import Http
+
+redis_con = redis.Redis(host='localhost', port=6379)
+redis_graph = Graph("apidoc", redis_con)
+"""
+getting the apidoc
+"""
+
+
+class _MemCache(dict):
+
+    def __nonzero__(self):
+        # even if empty, a _MemCache is True
+        return True
+
+    def set(self, key, value):
+        self[key] = value
+
+    def delete(self, key):
+        if key in self:
+            del self[key]
+
+
+DEFAULT_HTTP_CLIENT = Http(_MemCache())
+http = DEFAULT_HTTP_CLIENT
+APIDOC_RE = regex(
+    r'^<([^>]*)>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"$')
+
+
+def final_file(url):
+    response = urllib.urlopen(url)
+    new_file = json.loads(response.read())
+    return check_url(new_file, url)
+
+
+def check_url(new_file, url):
+    response, content = http.request(url, "GET")
+    link = response.get('link')
+    match = APIDOC_RE.match(link)
+    api_doc = match.groups()[0]
+    if api_doc != url:
+        print url, "not match", api_doc
+        new_url = api_doc
+        return final_file(new_url)
+    else:
+        return new_file
+
+
+url = "http://www.markus-lanthaler.com/hydra/event-api/"
+match_obj = re.match(r'(.*)://(.*)/(.*)/(.*)/(.*)/?', url, re.M | re.I)
+base_url = "{0}://{1}".format(match_obj.group(1), match_obj.group(2))
+entrypoint = "/" + match_obj.group(3) + "/" + match_obj.group(4) + "/"
+
+apidoc = final_file(url)
+
+"""
+extracting the endpoints and some of its properties or its api structure
+"""
+
+
+def get_endpoint_id(string):
+    return (entrypoint + string.replace("vocab:EntryPoint/", ""))
+
+
+def get_member_alias(string):
+    match_obj = re.match(r'/(.*)/(.*)/(.*)/(.*)?', string, re.M | re.I)
+#    base_url = "/{0}/{1}/".format(match_obj.group(1), match_obj.group(2))
+    entrypoint_member = "member" + match_obj.group(4)
+    return entrypoint_member
+
+
+def remove_vocab(string):
+    return string.replace("vocab:", "")
+
+
+""" Finding the entrypoint class for extracting the endpoints"""
```

Comment:
You did not have to do all the parsing for the API Documentation. Like I mentioned to you on the gitter chat, you can directly use the `hydrus.hydraspec.doc_maker.create_doc()` to parse the API Documentation. It will return an object of type `hydrus.hydraspec.doc_writer.HydraDoc`, you can directly access the endpoints, classes and their properties here as python objects, without having to search the dictionary for all that information. I remember asking you to read and understand the code so that you can reuse it.

Reply:
yeah I'll update it.